Mathlib/CategoryTheory/Sites/NonabelianCohomology/H1.lean

```diff
@@ -0,0 +1,206 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.Algebra.Category.GroupCat.Basic
+
+/-! The cohomology of a sheaf of groups in degree 1
+
+In this file, we shall define the cohomology in degree 1 of a sheaf
+of groups (TODO).
+
+Currently, given a presheaf of groups `G : Cᵒᵖ ⥤ GroupCat` and a family
+of objects `U : I → C`, we define 1-cochains/1-cocycles with values
+in `G` over `U`. (This definition neither requires the assumption that `G`
+is a sheaf, nor that `U` covers the terminal object.)
+
```

Comment:
Maybe it is worth adding some comment of the form "because we do not assume that the values of G are abelian, there is no general cohomology theory H^n, which is why we are only making definitions in low degree in this file".